dts/arm/atmel/saml2x.dtsi

```diff
@@ -57,146 +45,45 @@
 
 			flash0: flash@0 {
 				compatible = "soc-nv-flash";
-				reg = <0 0x40000>;
 				write-block-size = <4>;
 			};
 		};
 
-		pm: pm@40000400 {
-			compatible = "atmel,saml2x-pm";
-			reg = <0x40000400 0x400>;
-			interrupts = <0 0>;
-			#clock-cells = <2>;
-		};
-
-		mclk: mclk@40000400 {
-			compatible = "atmel,saml2x-mclk";
-			reg = <0x40000400 0x400>;
-			#clock-cells = <2>;
-		};
-
-		gclk: gclk@40001800 {
-			compatible = "atmel,saml2x-gclk";
-			reg = <0x40001800 0x400>;
-			#clock-cells = <1>;
-		};
-
-		dmac: dmac@44000400 {
-			compatible = "atmel,sam0-dmac";
-			reg = <0x44000400 0x400>;
-			interrupts = <5 0>;
-			#dma-cells = <2>;
+		sram0: memory@20000000 {
+			compatible = "mmio-sram";
+			reg = <0x20000000 0x8000>;
```

Comment:
Did you mean to omit the `reg = ...` line?

Reply:
RE: SAMRs and this comment

reg = being missing is relevant to the refactoring I did but also your concern of SAMR compatibility. There are multiple configurations of the SAML21 (and SAMR34) with different amounts of flash and sram. 

'saml21x18.dtsi' was the lowest level that memory size is "shared" across devices in this series and is why I added this file. 

It appears that my changes would require a similar addition to the samr34 tree.  
![image](https://github.com/zephyrproject-rtos/zephyr/assets/64241891/db904a4c-79a3-4ab6-a293-98f73818b23f)

I believe there's only one BSP that depends on the samr34/35 dtsi defs. It has the same memory configuration as defined in 'saml21x18' but would need an equivalent  'samr34x18' 

I'm not certain that samr35 is used currently but it would need a similar 'samr35x18' type dtsi. 

Thoughts on whether this hierarchy cleanup is worth pursing?